mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/GameScreen.kt

```diff
@@ -1,79 +1,149 @@
+            modifier =
+                modifier
+                    .verticalScroll(rememberScrollState())
+                    .padding(contentPadding)
+                    .padding(scaffoldPadding)
+                    .padding(start = 16.dp, end = 16.dp, top = 48.dp, bottom = 48.dp),
+            verticalArrangement = Arrangement.spacedBy(32.dp),
+        ) {
+          Column {
+            Player(White, state.white.name, state.white.message)
+            Player(Black, state.black.name, state.black.message)
+          }
+          ChessBoard(state)
+          Moves(state.moves, Modifier.fillMaxWidth())
+        }
+      },
+  )
 }
 
 /**
- * Display the list of moves that have been played
- * @param moves A list of [Move] that need to be displayed
- * @param modifier modifier the [Modifier] for the composable
+ * A composable which displays some basic information about a player. Each player may have a name,
+ * and an associated message.
+ *
+ * @param color the [ChessBoardState.Color] of the player.
+ * @param name the name of the player.
+ * @param message the message associated to the player.
+ * @param modifier the [Modifier] for this composable.
  */
 @Composable
-fun MoveList(moves: List<Move>, modifier: Modifier = Modifier) {
-  LazyRow(
-      horizontalArrangement = Arrangement.Start,
-      verticalAlignment = Alignment.CenterVertically,
-      modifier =
-          modifier
-              .padding(16.dp)
-              .border(
-                  width = 8.dp,
-                  color = MaterialTheme.colors.onPrimary,
-                  shape = RoundedCornerShape(16.dp)),
-  ) {
-    items(moves) { move ->
-      val color =
-          if (move.number % 2 == 1) MaterialTheme.colors.primary
-          else MaterialTheme.colors.primaryVariant
-      Text(
-          text = move.number.toString() + ". " + move.name,
-          color = color,
-          style = MaterialTheme.typography.subtitle1,
-          modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp))
+private fun Player(
+    color: ChessBoardState.Color,
+    name: String?,
+    message: Message,
+    modifier: Modifier = Modifier,
+) {
+  val green = if (color == White) Green500 else Green800
+  val icon = if (color == White) ChessIcons.WhiteKing else ChessIcons.BlackKing
+  Row(modifier, Arrangement.spacedBy(16.dp), CenterVertically) {
+    ProvideTextStyle(MaterialTheme.typography.subtitle1) {
+      CompositionLocalProvider(LocalContentColor provides green) {
+        Icon(icon, null, Modifier.size(24.dp))
+        Text(name ?: "")
+      }
+      Spacer(Modifier.weight(1f, fill = true))
+      CompositionLocalProvider(LocalContentColor provides Orange200) { Text(message.text) }
     }
   }
 }
 
+/** Retrieves the text associated with a [GameScreenState.Message]. */
+private val Message.text: String
+  @Composable
+  get() =
+      when (this) {
+        Message.None -> ""
+        Message.YourTurn -> LocalLocalizedStrings.current.gameMessageYourTurn
```

Comment:
Suggestion: Maybe you could also add a "Their turn" message, so that users don't think the game is bugged / stale.

Reply:
Excellent point ! I've added a comment on Figma so we keep track of this suggestion when we design a better UI to handle cases such as local games or spectating.